Azure/iotedge
# [Manual Cherry-Pick] [Master] Add E2E Test[Ubuntu AMD64] as Gate for Pull Requests

This PR is a manual port of the E2E Check-in Changes made in

https://github.com/Azure/iotedge/pull/5717
https://github.com/Azure/iotedge/pull/5743

Changes had to be manually ported over due to additional artifacts that get build for Master Branch

Deviation from #5717 
1. Manually Port Build Images  Instead of Cherry Picking since Master images.yaml builds builds more images than Release/1.2
2. Use Master Resources instead of release/1.2 Resources for downloading build artifacts

Testing
1. E2E Check-in Test run : https://dev.azure.com/msazure/One/_build/results?buildId=48267053&view=results
2. Build Images Pipeline: https://msazure.visualstudio.com/One/_build/results?buildId=48269740&view=results
3. Build Packages Pipeline: https://msazure.visualstudio.com/One/_build/results?buildId=48269877&view=results
4. CI E2E Test Run: https://dev.azure.com/msazure/One/_build/results?buildId=48272784&view=results

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_
